### DISCUSSION TITLE
Use Raspbian <=Buster

### DISCUSSION THREAD

**Discussion by xboxone08** [xboxone08/making_mcpi_real#2] [Announcements]
This project is a plugin for Minecraft: Pi Edition. The project would have worked fine on the latest Raspberry Pi OS Bullseye if there was a Minecraft: Pi Edition for that release. Minecraft: Pi Edition has been removed from the Raspberry Pi OS Bullseye apt repository. Buster is the latest version with Minecraft: Pi Edition, so that is what you need.

It's a real bummer that Minecraft: Pi Edition is no longer available on the latest release, but there's nothing I can do.

**Comment by Bigjango13** [comment]
But Minecraft: Pi Edition can still run on later versions of Raspbian, so the steps [here](https://askubuntu.com/a/648942) can be used to download and run it. Alternatively you could use [MCPI-Reborn](https://github.com/MCPI-Revival/minecraft-pi-reborn/blob/master/docs/INSTALL.md) which is a project that extends Minecraft: Pi Edition to add survival, servers, chat, nicer GUI, a easy way to mod it, and makes it work on different architectures along with many other things.

**Reply by NoozAbooz** [reply]
I believe MCPI was removed from the bullseye repos because there was some library incompatibility or something that prevented the game from working properly on bullseye

**Reply by xboxone08** [reply]
@Bigjango13 According to an answer on the [official Raspberry Pi forums](https://forums.raspberrypi.com/viewtopic.php?t=329601), Minecraft Pi is incompatible with Raspberry Pi OS Buster. The steps you listed don't list a version of Raspberry Pi OS, but if you tested it on Bullseye and it works, that is wonderful news.

MCPI-Reborn seems great, but I can't seem to find a list of what it does. Could you point me to that? Also, "an easy way to mod it" seems amazing. I know that it has survival and a nicer GUI (with PyKISS being an even nicer GUI?). When you say "makes it work on different architectures" do you mean I could run it on Windows?! This is amazing if so!!!

**Reply by Bigjango13** [reply]
> MCPI-Reborn seems great, but I can't seem to find a list of what it does. Could you point me to that? 

As far as I know it doesn't have an official feature list, but it does have a list of [feature flags](https://github.com/MCPI-Revival/minecraft-pi-reborn/blob/master/launcher/src/client/available-feature-flags) along with the default value. There are many things not mentioned that can be found in the [changelog](https://github.com/MCPI-Revival/minecraft-pi-reborn/blob/master/docs/CHANGELOG.md).

> Also, "an easy way to mod it" seems amazing. 

It is, and the next update is going to make it even easier! Mods can be made in [C/C++](https://wiki.mcpirevival.tk/wiki/Modding), but they can be used to add features for other languages to interact with. (Like [this mod](https://github.com/Bigjango13/MCPI-Addons) I made to extend the python API)

> When you say "makes it work on different architectures" do you mean I could run it on Windows?! This is amazing if so!!!

Sort of, it works on desktop computers running debian based distros (and may work on other distros, but it's buggy). There are way of making it work on windows (here's a [tutorial](https://youtu.be/3l-m8O13LYk)) but it isn't officially supported.

**Reply by xboxone08** [reply]
Wow! Thanks! They've been supporting Debian Bullseye for four versions! How would you normally mod MCPI, if possible?

**Reply by Bigjango13** [reply]
> How would you normally mod MCPI, if possible?

I don't think MCPI can be modded, however MCPI-Reborn has a wiki (thought it isn't updated as much as it should be) that has a page on modding it: [https://wiki.mcpirevival.tk/wiki/Modding](https://wiki.mcpirevival.tk/wiki/Modding)

**Reply by nobody5050** [reply]
I'd recommend joining the discord dedicated to our project if you want to ask questions without the github delay for answers, but we're also happy to keep answering here.

**Reply by xboxone08** [reply]
@nobody5050 OK, I did.

**Comment by notacoder23** [comment]
?
